File: Task 03/java/Subtask4/Task4.java
import java.io.FileReader;
import java.io.FileWriter;
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;

public class DiamondPatternToFile {
    public static void main(String[] args) {
        String inputFile = "input.txt";
        String outputFile = "output.txt"

        try {
            FileReader fr = new FileReader(inputFile);
            int n = 0;
            int c;
            StringBuilder number = new StringBuilder();


            while ((c = fr.read()) != -1) {
                if (Character.isDigit(c)) {
                    number.append((char) c);
                }
            }
            fr.close()
            n = Integer.parseInt(number.toString().trim());

            FileWriter fw = new FileWriter(outputFile);
            int mid = n / 2;

            for (int i = 0; i <= mid; i++) {
                for (int j = 0; j < mid - i; j++) {
                    fw.write(" ");
                }
                for (int j = 0; j < 2 * i + 1; j++) {
                    fw.write("*");
                }
                fw.write("\n");
            }
            for (int i = mid - 1; i >= 0; i--) {
                for (int j = 0; j < mid - i; j++) {
                    fw.write(" ");
                }
                for (int j = 0; j < 2 * i + 1; j++) {
                    fw.write("*");
                }
                fw.write("\n");
            }
            
            fw.close();
        } catch (IOException e) {
            System.err.println("An error occurred: " + e.getMessage());
        }
    }
}

File: Task 03/rust/Subtask4/Task4.rs
use std::fs::File;
use std::io::{self, Read, Write};

fn main() -> io::Result<()> {
    let mut input_file = File::open("input.txt")?;
    let mut input = String::new();
    input_file.read_to_string(&mut input)?;
    let n: usize = input.trim().parse().unwrap();

    let mut output_file = File::create("output.txt")?;

    for i in 0..n {
        for _ in 0..(n - i - 1) {
            write!(output_file, " ")?;
        }
        for _ in 0..(2 * i + 1) {
            write!(output_file, "*")?;
        }
        writeln!(output_file)?;
    }
    
    for i in (0..n - 1).rev() {
        for _ in 0..(n - i - 1) {
            write!(output_file, " ")?;
        }
        for _ in 0..(2 * i + 1) {
            write!(output_file, "*")?;
        }
        writeln!(output_file)?;
    }

    Ok(())
}
